projects/dsp-ui/src/lib/viewer/services/value-operation-event.service.ts

```diff
@@ -17,29 +17,38 @@ export class ValueOperationEventService {
 
     // Used in the listening component.
     // i.e. this.valueOperationEventSubscription = this._valueOperationEventService.on(Events.ValueAdded, () => doSomething());
-    on(event: Events, action: (newValue) => void): Subscription {
+    on(event: Events, action: (newValue: any) => void): Subscription {
         return this._subject$
             .pipe(
                 // Filter down based on event name to any events that are emitted out of the subject from the emit method below.
                 filter((e: EmitEvent) => e.name === event),
-                map((e: EmitEvent) => e.value)
+                map((e: EmitEvent) => e.value),
             )
             .subscribe(action); // Subscribe to the subject to get the data.
     }
 
     // Used in the emitting component.
-    // i.e. this.valueOperationEventService.emit(new EmitEvent(Events.ValueAdded));
+    // i.e. this.valueOperationEventService.emit(new EmitEvent(Events.ValueAdded, new EventValues(new ReadValue()));
     emit(event: EmitEvent) {
         this._subject$.next(event);
     }
 }
 
 export class EmitEvent {
-    constructor(public name: any, public value?: BaseValue) { }
+    constructor(public name: any, public value?: EventValues) { }
 }
 
 // Possible events that can be emitted.
 export enum Events {
     ValueAdded,
-    ValueDeleted
+    ValueDeleted,
+    ValueUpdated
+}
+
+/**
+ * @param currentValue the current value
+ * @param newValue value to update the current value with
```

Comment:
Could you explain why the current value would have to be updated with the new value?